Shared/MemoArtApp.swift

```diff
@@ -106,18 +109,40 @@ extension MemoArtApp {
 extension MemoArtApp {
     private static let highScoresKey = "MemoArtHighScores"
 
-    private static func loadHighScores() -> [HighScore] {
+    private static func loadHighScores() -> HighScoresState {
+        migrateHighScoresIfNeeded()
         guard
             let data = UserDefaults.standard.data(forKey: highScoresKey),
-            let highScores = try? JSONDecoder().decode([HighScore].self, from: data)
-        else { return [] }
+            let highScores = try? JSONDecoder().decode(HighScoresState.self, from: data)
+        else { return HighScoresState(boards: Boards(easy: [], normal: [], hard: [])) }
 
         return highScores
     }
 
-    private static func saveHighScores(highScores: [HighScore]) {
+    private static func saveHighScores(highScores: HighScoresState) {
         UserDefaults.standard.setValue(try? JSONEncoder().encode(highScores), forKey: highScoresKey)
     }
+
+    private static func migrateHighScoresIfNeeded() {
```

Comment:
This method should be removed at some point, let's say version 1.8.0 or 1.9.0

Reply:
I added a comment above the function, saying we should remove this code in version `1.9.0`